.codegen/service.go.tmpl

```diff
@@ -19,20 +19,42 @@ import (
 {{end}}
 
 {{define "service"}}
-var Cmd = &cobra.Command{
-	Use:   "{{(.TrimPrefix "account").KebabName}}",
-	{{- if .Description }}
-	Short: `{{.Summary | without "`"}}`,
-	Long: `{{.Comment  "  " 80 | without "`"}}`,
-	{{- end }}
-	Annotations: map[string]string{
-		"package": "{{ .Package.Name }}",
-	},
-	{{- if .IsPrivatePreview }}
+// Slice with functions to override default command behavior.
+// Functions can be added from the `init()` function in manually curated files in this directory.
+var cmdOverrides []func(*cobra.Command)
 
-	// This service is being previewed; hide from help output.
-	Hidden: true,
-	{{- end }}
+func New() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "{{(.TrimPrefix "account").KebabName}}",
+		{{- if .Description }}
+		Short: `{{.Summary | without "`"}}`,
+		Long: `{{.Comment  "  " 80 | without "`"}}`,
+		{{- end }}
+		GroupID: "{{ .Package.Name }}",
+		Annotations: map[string]string{
+			"package": "{{ .Package.Name }}",
+		},
+		{{- if .IsPrivatePreview }}
+
+		// This service is being previewed; hide from help output.
+		Hidden: true,
+		{{- end }}
+	}
+
+	{{ range .Methods }}
+	{{- $excludes := list "put-secret" -}}
+	{{- if in $excludes .KebabName -}}
+		{{ continue }}
+	{{- end -}}
+	cmd.AddCommand(new{{.PascalName}}())
+	{{ end }}
```

Comment:
This is the same as the logic from lines 38-44 in the original. Perhaps we can at least reuse the same `$excludes` variable rather than redefining it here?

Reply:
Good point, investigating how we can best do that.

Reply:
I read https://pkg.go.dev/text/template@go1.20.6 and was not able to find a way to reuse the contents of a block as a parameter itself without making it much less readable (perhaps you could define a template for the inner block and pass its name as a parameter to the range template but I didn't try).

That said, I was able to remove the reuse entirely by adding an `init()` function for each command to add it to the list of subcommand for the command group.

Reply:
Nice. BTW: one way this could be done is by adding it to the OpenAPI code generator. We may find it useful to define something like a variable map in .codegen.json so that we don't need to change the code generator to introduce new parameters like this.